Bbk req update (#330)
w3c/secure-payment-confirmation: bbk-requirements.md

## Assumptions

* Device binding is the fundamental property of a BBK. Once a BBK has been associated with a device, it cannot be reused outside that device.

> * Although in theory a BBK could be shared among multiple [installed browser programs](#installed-browser-program) on the same device, we assume here that a BBK will also be bound to an [installed browser program](#installed-browser-program) (as the name "browser bound key" indicates).

Reviewer: This assumption sticks out to me on a re-read, @ianbjacobs . Do you know what we're trying to say here by "Although in theory a BBK could be shared [..]"?

Is it not a **requirement** rather than an **assumption** that BBKs are bound to a single installed browser program?

Reviewer 2: I agree that with this recent round of clarifications, we have elevated this to a requirement. I think we can delete this bullet.

Author: Makes sense, I'll remove it

* The consensus of the Web Payments Working Group is that, initially, each BBK will be paired with a passkey. We assume here that a BBK will be bound to a single passkey. In the future the Web Payments Working Group may explore the use of BBKs without passkeys.
* A given passkey corresponds to an online identity. To distinguish online identities, the user will use different passkeys. 
* A relying party will perform some ID &amp; V process before trusting a (new) BBK. That ID &amp; V process might take place before a Web Authentication registration (and thus, if the BBK is returned as part of the Web Authentication registration, the RP would not likely step up the user a second time). In the case of a synced passkey, when the RP first sees a BBK on a new user device, in the absence of other trust signals, the RP would likely perform some ID &amp; V process in order to trust the new BBK, and we consider that an acceptable user experience on a new device.

## Definitions

* <dfn id="bbk-binding">BBK binding</dfn>: a unique triplet consisting of a specific passkey, a specific [installed browser program](#installed-browser-program), and a specific device.

> * <dfn id="installed-browser-program">Installed browser program</dfn>: a specific installation of a web browser on a device. Note that a browser embedded in another application (e.g. WebView)  is considered a part of the embedding application and thus a distinct installed browser program.

Reviewer 2: ```suggestion
* <dfn id="installed-browser-program">Installed browser program</dfn>: a specific installation of a web browser on a device. Note that a browser embedded in another application (e.g., WebView)  is considered a part of the embedding application and thus a distinct installed browser program.
```

Reviewer 2: I think we should put the definition of Installed browser program before the definition of BBK binding (since the latter uses the former).

* This proposal relies on underlying [FIDO security assumptions](https://fidoalliance.org/specs/common-specs/fido-security-ref-v2.1-ps-20220523.html#fido-security-assumptions) and related threat models.

## FAQ
* Can a BBK be reused across different accounts on the same installed browser program?

>   * Implementation specific.

Reviewer 2: ```suggestion
  * This is an implementation choice left to the installed browser program.
```

> * Can a BBK be reused across an installed browser program and a web view version of that browser?

Reviewer 2: ```suggestion
* Can a BBK be reused across an installed browser program and a WebView version of that browser?
```

>   * No; each webview is considered a distinct installed browser program bounded by the embedding application.

Reviewer 2: ```suggestion
  * No; each WebView is considered a distinct installed browser program bounded by the embedding application.
```

> * Can a BBK be reused across an installed browser program and other installed browser programs using the same underlying engine (eg. Chromium)

Reviewer 2: ```suggestion
* Can a BBK be reused across an installed browser program and other installed browser programs using the same underlying engine (e.g., Chromium)?
```
